nx_gzip/nx_gzip.py

```diff
@@ -0,0 +1,471 @@
+
+    def test_zlib_series(self):
+        """
+        Running NX-GZIP: Run compress/decompress zlib series
+        """
+        self.log.info("NX-GZIP: test_zlib_series:\
+                      Run compress/decomp zlib test series")
+
+        test_dir = '%s/samples' % (self.sourcedir)
+        os.chdir(test_dir)
+        failed_tests = []
+        output = build.run_make(
+            test_dir, extra_args='bench',
+            process_kwargs={"ignore_status": True})
+        for line in output.stdout.decode('utf-8').splitlines():
+            if "failed" in line:
+                failed_tests.append(line)
+        if failed_tests:
+            self.fail("%s" % failed_tests)
+
+        file_size = self.params.get('file_size', default='5')
+        dd_cmd = 'dd if=/dev/urandom of=%sgb-file bs=1000000000 count=%s'\
+                 % (file_size, file_size)
+        if process.system(dd_cmd):
+            self.fail("NX-GZIP: test_zlib_series: dd file creation failed")
+
+        zlib_cmd = './zlib-run-series.sh %sgb-file' % file_size
+        if process.system(zlib_cmd, shell=True):
+            self.fail("NX-GZIP: test_zlib_series: zlib test series failed")
+
+        self.log.info("remove %sgb-file file" % file_size)
+        if os.remove("%sgb-file" % file_size):
+            self.fail("NX-GZIP: test_zlib_series: %sgb-file remove failed"
+                      % file_size)
+
+    def test_compdecomp_2nx(self):
+        """
+        Running NX-GZIP: Run compress/decompress on 2nx devices
+        """
+        self.log.info("NX-GZIP: test_compdecomp_2nx:\
+                      Run compress/decompress on 2nx devices")
+
+        test_dir = '%s/samples' % (self.sourcedir)
+        os.chdir(test_dir)
+        failed_tests = []
+        output = build.run_make(
+            test_dir, extra_args='bench',
+            process_kwargs={"ignore_status": True})
+        for line in output.stdout.decode('utf-8').splitlines():
+            if "failed" in line:
+                failed_tests.append(line)
+        if failed_tests:
+            self.fail("%s" % failed_tests)
+
+        file_size = self.params.get('file_size', default='5')
+        dd_cmd = 'dd if=/dev/urandom of=%sgb-file bs=1000000000 count=%s'\
+                 % (file_size, file_size)
+        if process.system(dd_cmd):
+            self.fail("NX-GZIP: test_compdecomp_2nx: dd file creation failed")
+
+        nx2_cmd = './run-series_2nx.sh %sgb-file' % file_size
+        if process.system(nx2_cmd, shell=True):
+            self.fail("NX-GZIP: test_compdecomp_2nx:\
+                      comp/decomp on 2nx devices tests failed")
+
+        self.log.info("remove %sgb-file file" % file_size)
+        if os.remove("%sgb-file" % file_size):
+            self.fail("NX-GZIP: test_compdecomp_2nx:\
+                      %sgb-file remove failed" % file_size)
+
+    def tearDown(self):
+        if self.name.uid == 5 or self.name.uid == 7 or self.name.uid == 9\
```

Comment:
better not to hardcore the test uid values here. In any case this line can be removed because the checks following this will ensure removing of directory irrespective of the test executed, which should be fine here.